core/src/types/extended-repositories/repository.ts

```diff
@@ -0,0 +1,69 @@
+import type {DataSource, DeepPartial, FindManyOptions, FindOneOptions, SelectQueryBuilder} from "typeorm";
+import {Repository} from "typeorm";
+
+import type {BaseEntity} from "../base-entity";
+import type {Class} from "./repository.types";
+
+export abstract class ExtendedRepository<T extends BaseEntity> extends Repository<T> {
+    constructor(readonly c: Class<T>, readonly dataSource: DataSource) {
+        super(c, dataSource.createEntityManager());
+    }
+
+    async get(options: FindOneOptions<T>): Promise<T> {
+        return this.findOneOrFail(options);
+    }
+
+    async getOrNull(options: FindOneOptions<T>): Promise<T | null> {
+        return this.findOne(options);
+    }
```

Comment:
These are essentially just aliases over the `Repository` methods, I think we can scrap them and just use `findOneOrFail` and `findOne` directly